app/src/main/java/org/simple/clinic/login/TemporaryValidateOtpRequestInterceptor.kt

```diff
@@ -0,0 +1,65 @@
+package org.simple.clinic.login
+
+import com.squareup.moshi.Moshi
+import okhttp3.Interceptor
+import okhttp3.MediaType
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.ResponseBody
+import okio.Buffer
+import org.simple.clinic.user.LoggedInUser
+import org.simple.clinic.user.LoggedInUserPayload
+import org.simple.clinic.user.UserStatus
+import org.threeten.bp.Instant
+import java.nio.charset.Charset
+import java.util.UUID
+import javax.inject.Inject
+
+// TODO: Integrate APIs and remove this class when they are ready
+class TemporaryValidateOtpRequestInterceptor @Inject constructor(
+    private val moshi: Moshi
+) : Interceptor {
+
+  override fun intercept(chain: Interceptor.Chain): Response {
+    val originalRequest = chain.request()
+
+    val urlString = originalRequest.url().toString()
+
+    return when {
+      urlString.endsWith("v1/requestOtp") -> tempSendOtpResponse(originalRequest)
+      urlString.endsWith("v1/validateLoginOtp") -> tempValidateLoginOtp(originalRequest)
+      else -> chain.proceed(originalRequest)
+    }
+  }
+
+  private fun tempValidateLoginOtp(request: Request): Response {
+    // Some sleep to simulate a real network call
+    Thread.sleep(1000L)
+    val buffer = Buffer()
+    request.body()!!.writeTo(buffer)
+    val validateLoginOtpRequest = moshi.adapter(ValidateLoginOtpRequest::class.java).fromJson(buffer.readString(Charset.forName("UTF-8")))!!
+    val loginResponse = LoginResponse(
+        accessToken = "access_token",
+        loggedInUser = LoggedInUserPayload(
+            UUID.fromString(validateLoginOtpRequest.userId),
+            fullName = "",
+            phoneNumber = "",
+            pinDigest = validateLoginOtpRequest.passwordDigest,
+            facilityUuids = listOf(UUID.randomUUID()),
+            createdAt = Instant.now(),
+            updatedAt = Instant.now(),
+            status = UserStatus.DISAPPROVED_FOR_SYNCING
+        )
+    )
+
+    return Response.Builder().code(200).body(ResponseBody.create(MediaType.parse("application/json"), moshi.adapter(LoginResponse::class.java)
+        .toJson(loginResponse))).build()
```

Comment:
I want to ask you to format your code even if it's temporary because we're working on an open source project and this will be visible to people outside. There are not a lot of eyes right now, but we will get there soon.